src/ui/Interface.js

```diff
@@ -19,8 +19,20 @@ module.exports = function Interface (options) {
 
   panMapWhenInputsChange();
 
+  function onDrag() {
+    function changeVal(result) {
+      $("#location").val(result.results[0].formatted_address);
+    }
+    options.map.on('moveend', function(e) {
+     options.getPlacenameFromCoordinates(options.getLat(), options.getLon(), changeVal)
```

Comment:
Actually the content of this function IS onDrag, and the event listener should be set up in the construction of the class, on startup. That way you can set the behavior that runs upon dragging, as an option. The default should be what's here, but you can override it via the options. Does that make sense? So `options.onDrag = options.onDrag || function onDrag(event) { ...this line here... }`